plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -894,7 +900,9 @@ public synchronized void addPartition(
             String tableName,
             Partition partition,
             Path currentLocation,
-            PartitionStatistics statistics)
+            Optional<List<String>> files,
```

Comment:
Same question here, why does it have to be optional? Should it be set unconditionally (then it would be possible to make the fileNames filed in the `PartitionAndMore` non Optional)

Reply:
Again. Optional is more explicit. It stresses the intent that we do not care about the list of files here. I can change. But I would rather not to.